game-core/src/main/java/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -360,67 +361,61 @@ private void createAndSaveStats(final boolean showPhaseStats) {
-                  defaultFileName = defaultFileName + ".html";
-                  chooser.setSelectedFile(new File(rootDir, defaultFileName));
-                  if (chooser.showSaveDialog(frame) != JOptionPane.OK_OPTION) {
-                    return;
-                  }
-                  try (Writer writer =
-                      Files.newBufferedWriter(
-                          chooser.getSelectedFile().toPath(), StandardCharsets.UTF_8)) {
-                    writer.write(
-                        HelpMenu.getUnitStatsTable(gameData, uiContext)
-                            .replaceAll("</?p>|</tr>", "$0\r\n")
-                            .replaceAll("(?i)<img[^>]+/>", ""));
-                  } catch (final IOException e1) {
-                    log.log(
-                        Level.SEVERE,
-                        "Failed to write unit stats: "
-                            + chooser.getSelectedFile().getAbsolutePath(),
-                        e1);
-                  }
-                }));
-    menuFileExport.setMnemonic(KeyEvent.VK_U);
-    add(menuFileExport);
+  private JMenuItem createExportUnitStatsMenu() {
+    return new JMenuItemBuilder("Export Unit Charts", KeyCode.U)
+        .actionListener(this::exportUnitCharts)
+        .build();
   }
 
-  private void addExportSetupCharts() {
-    final JMenuItem menuFileExport =
-        new JMenuItem(
-            SwingAction.of(
-                "Export Setup Charts",
-                e -> {
-                  final JFrame frame = new JFrame("Export Setup Charts");
-                  frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
-                  final GameData clonedGameData;
-                  gameData.acquireReadLock();
-                  try {
-                    clonedGameData = GameDataUtils.cloneGameData(gameData);
-                  } finally {
-                    gameData.releaseReadLock();
-                  }
-                  final JComponent newContentPane = new SetupFrame(clonedGameData);
-                  // content panes must be opaque
-                  newContentPane.setOpaque(true);
-                  frame.setContentPane(newContentPane);
-                  // Display the window.
-                  frame.pack();
-                  frame.setLocationRelativeTo(frame);
-                  frame.setVisible(true);
-                  uiContext.addShutdownWindow(frame);
-                }));
-    menuFileExport.setMnemonic(KeyEvent.VK_C);
-    add(menuFileExport);
+  private void exportUnitCharts() {
+    final JFileChooser chooser = new JFileChooser();
+    chooser.setFileSelectionMode(JFileChooser.FILES_ONLY);
+    final File rootDir = new File(SystemProperties.getUserDir());
+    String defaultFileName = gameData.getGameName() + "_unit_stats";
+    defaultFileName = FileNameUtils.removeIllegalCharacters(defaultFileName);
+    defaultFileName = defaultFileName + ".html";
```

Comment:
Side-note for a future PR:
Building this string can really be simplified to
```java
final String defaultFileName = FileNameUtils.removeIllegalCharacters(gameData.getGameName() + "_unit_stats") + ".html";
```
or maybe
```java
final String defaultFileName = FileNameUtils.removeIllegalCharacters(gameData.getGameName()) + "_unit_stats.html";
```
even